### PR TITLE
Feature/st 144 visa checkout customizations

### PR DESCRIPTION
**Pull request by james-simpson-ST**

### CODE REVIEW

File: src/core/classes/VisaCheckout.ts [hunk 1]
```diff
@@ -45,31 +47,73 @@ class VisaCheckout {
...
+      config = { ...config,
+                ...settings 
+               };
+    }
+    return config;
+  }
+
+  public _setButtonSettings(settings: any, stJwt: StJwt) {
+    let config = {locale: stJwt.locale};
+    if (settings != undefined) {
```
Comment (pgs-gmartowski) [inline src/core/classes/VisaCheckout.ts]: I think we can avoid code duplication in this case by adding function which will be look more, less like:
`private setConfiguration = (config, settings) =>  settings ? {...config, ...settings} : ''  `

File: src/core/shared/Utils.ts [hunk 2]
```diff
@@ -42,6 +42,22 @@ export default class Utils {
...
+   * @param params The parameters as key, value pairs with which to update the URL
+   * @return The new URL including the extra parameters
+   */
+  public static addUrlParams(url: string, params: any) {
+    if (params != undefined) {
+      let urlParams = new URLSearchParams();
+      for (let param in params) {
+        urlParams.set(param, params);
+      }
+      return url + "?" + urlParams.toString();
```
Comment (pgs-gmartowski) [inline src/core/shared/Utils.ts]: i'm not sure but it seems like we can use template strings here: 
`return `${url}?${urlParams.toString()}`;`

Comment (aaron-moore-ST) [reply src/core/shared/Utils.ts]: As mentioned in pr#83 can we use the URL interface? That would be a sturdier method to build URLs if we can.

Comment (james-simpson-ST) [reply src/core/shared/Utils.ts]: I tried to use URL but this requires the URL to contain the full domain and path which I didn't want to hard code the Domain. This is good enough for the usecase and I've updated to use template strings. The only case that does not work is if the URL already contains parameters which for now we do not have this case therefore going with YAGNI we'll add that when we need it

File: src/core/classes/VisaCheckout.ts (hunk 1 above)
Comment (james-simpson-ST) [reply src/core/classes/VisaCheckout.ts]: Done

File: src/core/shared/Utils.ts (hunk 2 above)
Comment (pgs-kzasada) [reply src/core/shared/Utils.ts]: `URLSearchParams` won't work in IE. I'll look for a workaround. Moreover, using `new URLSearchParams()` you can pass an object with params as an argument, so `for` loop is not needed.

File: src/core/shared/StJwt.ts [hunk 3]
```diff
@@ -44,6 +44,18 @@ export class StJwt {
...
+    /**
+    * Convenience getter for returning the locale
+    * @return The locale contained inside the encoded payload or en_GB if undefined
+    */
+    public get locale() {
+        let locale = "en_GB";
+        if (this.payload.locale) {
+            locale = this.payload.locale;
+        }
+        return locale;
```
Comment (pgs-kzasada) [inline src/core/shared/StJwt.ts]: Minor thing, but you can simplify to `return this.payload.locale || 'en_GB'`. Are you sure that `this.payload` will be always defined?

File: src/core/shared/Utils.ts [hunk 4]
```diff
@@ -42,6 +42,22 @@ export default class Utils {
...
 
+  /**
+   * Helper function to return a URL string which has parameters added to it
+   * @param url The current URL as a string
+   * @param params The parameters as key, value pairs with which to update the URL
+   * @return The new URL including the extra parameters
+   */
+  public static addUrlParams(url: string, params: any) {
+    if (params != undefined) {
+      let urlParams = new URLSearchParams();
```
Comment (pgs-kzasada) [inline src/core/shared/Utils.ts]: If `params` is an object you can pass this object as an argument to `new URLSearchParams` and for loop won't be needed.

File: src/core/shared/StJwt.ts (hunk 3 above)
Comment (james-simpson-ST) [reply src/core/shared/StJwt.ts]: Done (yes this.payload should always exist at this point)

File: src/core/shared/Utils.ts (hunk 4 above)
Comment (james-simpson-ST) [reply src/core/shared/Utils.ts]: Done

**Pull request merged**